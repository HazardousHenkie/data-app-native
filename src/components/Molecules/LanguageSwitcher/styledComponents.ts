import styled from 'styled-components'

import { View } from 'react-native'

const LanguageSwitcherWrapper = styled(View)`
    position: relative;
    width: 100%;
`

export default LanguageSwitcherWrapper
