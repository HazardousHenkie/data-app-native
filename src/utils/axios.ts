import axios from 'axios'

const Axios = axios.create({
    baseURL: process.env.REACT_APP_API_URL,
})

Axios.defaults.headers.common['Content-Type'] = 'application/json;charset=utf-8'

export default Axios
