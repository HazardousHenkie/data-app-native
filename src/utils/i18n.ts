import i18n from 'i18next'
import * as Localization from 'expo-localization'

import { initReactI18next } from 'react-i18next'

import translationEN from '../../translations/locales/en.json'
import translationJA from '../../translations/locales/ja.json'
import translationNL from '../../translations/locales/nl.json'

const resources = {
    en: translationEN,
    ja: translationJA,
    nl: translationNL,
}

export const appLocales = ['en', 'ja', 'nl']

i18n.use(initReactI18next).init({
    resources,
    debug: true,
    lng: Localization.locale,
    fallbackLng: 'en',
    whitelist: appLocales,
    interpolation: {
        escapeValue: false,
    },
    react: {
        useSuspense: false,
        wait: true,
    },
})

export default i18n
