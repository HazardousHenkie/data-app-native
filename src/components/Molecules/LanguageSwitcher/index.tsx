import React, { useState, useEffect } from 'react'

import { useTranslation } from 'react-i18next'

import RNPickerSelect from 'react-native-picker-select'

import LanguageSwitcherWrapper from './styledComponents'
import { appLocales } from '../../../utils/i18n'

const LanguageSwitcher: React.FC = () => {
    const [loadingLanguage, setloadingLanguage] = useState(false)
    const { i18n } = useTranslation()
    const dropdownItems = appLocales.map((value) => ({ value, label: value }))

    const onLocaleToggle = (itemValue: string) => {
        setloadingLanguage(true)
        i18n.changeLanguage(itemValue as string)
    }

    useEffect(() => {
        setloadingLanguage(false)
    }, [i18n.language])

    return (
        <>
            <LanguageSwitcherWrapper data-testid="languageSwitcher">
                <RNPickerSelect
                    placeholder={{}}
                    value={i18n.language}
                    onValueChange={onLocaleToggle}
                    disabled={loadingLanguage}
                    items={dropdownItems}
                />
            </LanguageSwitcherWrapper>
        </>
    )
}

export default LanguageSwitcher
