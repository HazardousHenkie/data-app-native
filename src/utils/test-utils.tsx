import React from 'react'
import { render } from '@testing-library/react-native'

// import { Provider } from 'react-redux'

import { ThemeProvider } from 'styled-components'
import { Provider as PaperProvider } from 'react-native-paper'

import { I18nextProvider } from 'react-i18next'

import { NavigationContainer } from '@react-navigation/native'
import i18n from './i18n'

import darkTheme from '../styles/themeStyles'
import variables from '../styles/variables'

// import configureStore from '../store/configureStore'

// TODO: add store back to test when there is a reducer

const customRender = (
    ui: React.ReactElement,
    {
        // initialState = {},
        // store = configureStore(initialState),
        ...renderOptions
    } = {}
) => {
    const Wrapper: React.FC = ({ children }) => {
        return (
            // <Provider store={store}>
            <NavigationContainer>
                <I18nextProvider i18n={i18n}>
                    <ThemeProvider theme={variables}>
                        <PaperProvider theme={darkTheme}>
                            {children}
                        </PaperProvider>
                    </ThemeProvider>
                </I18nextProvider>
            </NavigationContainer>
            // </Provider>
        )
    }

    return render(ui, {
        wrapper: Wrapper,
        ...renderOptions,
    })
}

export * from '@testing-library/react-native'

export { customRender as render }
