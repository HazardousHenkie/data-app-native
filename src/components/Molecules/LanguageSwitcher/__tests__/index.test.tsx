import React from 'react'

import { render } from '../../../../utils/test-utils'

import LanguageSwitcher from '../index'

describe('<LanguageSwitcher />', () => {
    it('Should render like snapshot', () => {
        const component = render(<LanguageSwitcher />)
        expect(component).toMatchSnapshot()
    })

    // it("gets initial language and it's being passed", () => {
    //     const { getByText } = render(<LanguageSwitcher />)
    //     const selectInput = getByText('en')
    //     console.log(selectInput)
    //     expect(selectInput).toHaveProperty('value', 'en')
    // })

    // it('Check if language changes after select', () => {
    //     const { getByRole, container } = render(<LanguageSwitcher />)
    //     fireEvent.mouseDown(getByRole('button'))
    //     const listbox = within(getByRole('listbox'))
    //     fireEvent.click(listbox.getByText('ja'))
    //     expect(container.querySelector('input')).toHaveProperty('value', 'ja')
    //     expect(getByRole('button')).toHaveTextContent('ja')
    // })
    // it('Check listitems', () => {
    //     const { getByRole, getAllByRole } = render(<LanguageSwitcher />)
    //     fireEvent.mouseDown(getByRole('button'))
    //     const listItems = getAllByRole('option')
    //     expect(listItems).toHaveLength(3)
    // })
})
