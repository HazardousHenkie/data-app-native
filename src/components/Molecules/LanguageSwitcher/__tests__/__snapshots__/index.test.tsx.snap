// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LanguageSwitcher /> Should render like snapshot 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    className="sc-bdfBwQ jSBXlJ"
    testID="languageSwitcher"
  >
    <View
      style={
        Array [
          Object {
            "alignSelf": "stretch",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="ios_touchable_wrapper"
      >
        <View
          pointerEvents="box-only"
        >
          <TextInput
            allowFontScaling={true}
            editable={false}
            rejectResponderTermination={true}
            style={
              Array [
                undefined,
                Object {},
              ]
            }
            testID="text_input"
            underlineColorAndroid="transparent"
            value="en"
          />
        </View>
      </View>
      <Modal
        hardwareAccelerated={false}
        onOrientationChange={[Function]}
        supportedOrientations={
          Array [
            "portrait",
            "landscape",
          ]
        }
        testID="ios_modal"
        transparent={true}
        visible={false}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "opacity": 1,
            }
          }
          testID="ios_modal_top"
        />
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#f8f8f8",
                "borderTopColor": "#dedede",
                "borderTopWidth": 1,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "space-between",
                "paddingHorizontal": 10,
                "zIndex": 2,
              },
              undefined,
            ]
          }
          testID="input_accessory_view"
        >
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                undefined,
              ]
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderColor": "#a1a1a1",
                      "borderRightWidth": 1.5,
                      "borderTopWidth": 1.5,
                      "height": 15,
                      "width": 15,
                    },
                    undefined,
                    Object {
                      "marginLeft": 11,
                      "transform": Array [
                        Object {
                          "translateY": 4,
                        },
                        Object {
                          "rotate": "-45deg",
                        },
                      ],
                    },
                    undefined,
                    Object {},
                  ]
                }
              />
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderColor": "#a1a1a1",
                      "borderRightWidth": 1.5,
                      "borderTopWidth": 1.5,
                      "height": 15,
                      "width": 15,
                    },
                    undefined,
                    Object {
                      "marginLeft": 22,
                      "transform": Array [
                        Object {
                          "translateY": -5,
                        },
                        Object {
                          "rotate": "135deg",
                        },
                      ],
                    },
                    undefined,
                    Object {},
                  ]
                }
              />
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            hitSlop={
              Object {
                "bottom": 4,
                "left": 4,
                "right": 4,
                "top": 4,
              }
            }
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="done_button"
          >
            <View
              testID="needed_for_touchable"
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#007aff",
                      "fontSize": 17,
                      "fontWeight": "600",
                      "paddingRight": 11,
                      "paddingTop": 1,
                    },
                    undefined,
                    Object {},
                  ]
                }
                testID="done_text"
              >
                Done
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#d0d4da",
                "justifyContent": "center",
              },
              Object {
                "height": 215,
              },
              undefined,
            ]
          }
        >
          <View>
            <RNCPicker
              items={
                Array [
                  Object {
                    "label": "en",
                    "textColor": undefined,
                    "value": "en",
                  },
                  Object {
                    "label": "ja",
                    "textColor": undefined,
                    "value": "ja",
                  },
                  Object {
                    "label": "nl",
                    "textColor": undefined,
                    "value": "nl",
                  },
                ]
              }
              onChange={[Function]}
              selectedIndex={0}
              style={
                Array [
                  Object {
                    "height": 216,
                  },
                  undefined,
                ]
              }
              testID="ios_picker"
            />
          </View>
        </View>
      </Modal>
    </View>
  </View>
</View>
`;
